### PR TITLE
Feat(Segmentation): add losses

### PR DESCRIPTION
**Pull request by jdeschamps**
Add semantic segmentation losses.

This PR adds `DiceLoss`, `CrossEntropyLoss` and `DiceCELoss` (Dice+CrossEntropy) losses. All losses assume that the input to the `forward` method are:
- `inputs`: raw model output as logits of dims (B, C, [Z], Y, X) where `C` has dimension `num_classes` (including background)
- `targets`: class labels with singleton channel of dims (B, 1, [Z], Y, X).

Dice loss converts the input to probabilities (softmax) and the targets to one-hot encoding, while cross-entropy simply removes from the target the singleton `C` dimension and converts to `long` (necessary for `torch.nn.functional.cross_entropy`).

### Added

- Segmentation losses to `segmentation_losses`
- Corresponding tests

## Notes

Typing added in https://github.com/CAREamics/careamics/pull/1081.

Part of splitting https://github.com/CAREamics/careamics/pull/986 into smaller PRs.

### CODE REVIEW

**Comment by jdeschamps** [comment]
Note: this PR was merged in #1068 as it was needed there, then it is modified by #1081.

**Comment by jdeschamps** [comment]
Superseded by https://github.com/CAREamics/careamics/pull/1068
